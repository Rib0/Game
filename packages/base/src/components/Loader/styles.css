@keyframes loading {
    100% {
        transform: translate(-50%, -50%) rotate(360deg);
    }
}

.loadingContainer {
    position: fixed;
    top: 0;
    bottom: 0;
    left: 0;
    right: 0;
    opacity: 0.3;
    background: gray;
    z-index: 222223;
}

.loader {
    position: fixed;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    width: 70px;
    height: 70px;
    border-radius: 50%;
    border: 12px solid #fff;
    border-top-color: transparent;
    animation: loading 2s linear infinite;
}
