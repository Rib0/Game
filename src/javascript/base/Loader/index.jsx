import React from 'react';

import styles from './styles.css';

const Loading = () => (
    <>
        <div className={styles.loadingContainer} />
        <div className={styles.loader} />
    </>
);

export default Loading;
