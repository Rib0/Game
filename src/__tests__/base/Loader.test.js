import React from 'react';
import Loader from 'base/Loader';

const setUp = () => shallow(<Loader />);

describe('Loader test', () => {
    test('Loader should render', () => {
        const Component = setUp();
        expect(Component).toMatchSnapshot();
    });
    test('Loader should have 2 divs', () => {
        const Component = setUp();
        expect(Component.find('div')).toHaveLength(2);
    });
});
