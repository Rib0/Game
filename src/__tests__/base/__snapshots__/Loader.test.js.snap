// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader test Loader should render 1`] = `
<Fragment>
  <div
    className="loadingContainer"
  />
  <div
    className="loader"
  />
</Fragment>
`;
